src/main/java/org/ohdsi/circe/cohortdefinition/VisitDetail.java

```diff
@@ -41,44 +41,44 @@ public class VisitDetail extends Criteria {
   @JsonProperty("VisitDetailEndDate")
   public DateRange visitDetailEndDate;
 
-  @JsonProperty("VisitDetailType")
-  public Concept[] visitDetailType;
+  @JsonProperty("VisitDetailTypeCS")
+  public Integer visitDetailTypeCS;
 
   @JsonProperty("VisitDetailTypeExclude")
   public boolean visitDetailTypeExclude = false;
 
-  @JsonProperty("VisitDetailSourceConcept")
-  public Integer visitDetailSourceConcept;
+  @JsonProperty("VisitDetailSourceConceptCS")
+  public Integer visitDetailSourceConceptCS;
 
   @JsonProperty("VisitDetailLength")
   public NumericRange visitDetailLength;
   
   @JsonProperty("Age")
   public NumericRange age;
   
-  @JsonProperty("Gender")
-  public Concept[] gender;
+  @JsonProperty("GenderCS")
+  public Integer genderCS;
   
-  @JsonProperty("ProviderSpecialty")
-  public Concept[] providerSpecialty;
+  @JsonProperty("ProviderSpecialtyCS")
+  public Integer providerSpecialtyCS;
 
-  @JsonProperty("PlaceOfService")
-  public Concept[] placeOfService;
+  @JsonProperty("PlaceOfServiceCS")
+  public Integer placeOfServiceCS;
 
-  @JsonProperty("DischargedToConcept")
-  public Concept[] dischargedToConcept;
+  @JsonProperty("DischargedToConceptCS")
+  public Integer dischargedToConceptCS;
 
-  @JsonProperty("AdmittedFromConcept")
-  public Concept[] admittedFromConcept;
+  @JsonProperty("AdmittedFromConceptCS")
+  public Integer admittedFromConceptCS;
 
   /**
    * ID of Codeset which defines Geo concepts.
    * The care site's location.region_concept_id should match one of those.
    */
 
   @CdmVersion(range = ">=6.1")
-  @JsonProperty("PlaceOfServiceLocation")
-  public Integer placeOfServiceLocation;
+  @JsonProperty("PlaceOfServiceLocationCS")
+  public Integer placeOfServiceLocationCS;
```

Comment:
This is already a concept set, so we don't need to change this to use a CS.   VisitOccurrence already has a `PlaceOfServiceLocation` property that is an Integer, so we will keep this consistent in VisitDetail.  You can revert this change back to:

```
  @JsonProperty("PlaceOfServiceLocation")
  public Integer placeOfServiceLocation;
```

Reply:
fixed